playbooks/resize_disk.yaml

```diff
@@ -0,0 +1,41 @@
+---
+- hosts: all
+  become: true
+  tasks:
+    - name: Find root partition
+      shell: df -hT | grep /$
+      register: root_partition
+
+    - name: Install cloud-utils-growpart package
+      package:
+        name: cloud-utils-growpart
+      when: ansible_os_family == 'RedHat'
+
+    - name: Install cloud-guest-utils package
+      package:
+        name: cloud-guest-utils
+      when: ansible_os_family == 'Debian'
+
+    - name: Install e2fsprogs package
+      package:
+        name: e2fsprogs
+      when: '"ext4" in root_partition.stdout'
+
+    - name: Extend partition
+      command: growpart /dev/vda {{ root_partition.stdout.split(' ')[0].replace('/dev/vda', '') }}
+      register: output
+      ignore_errors: true
+
+    - name: Fail if growpart fails, do not fail if NOCHANGE
+      fail:
+      when:
+        - '"NOCHANGE" not in output.stdout'
+        - output.rc != 0
```

Comment:
Does't this make the `vagrant provision` command fail on runs after the first one?

Reply:
Nope. This is an `and` condition. So if it reports a non-zero return code but does not contain NOCHANGE in the output, then the error is likely a real error and we should fail. For whatever reason, NOCHANGE causes a return code of `1` and from all my testing the same is true if oyu attempt it with `--dry-run`.